packages/twenty-front/src/modules/workflow/search-variables/components/VariableTagInput.tsx

```diff
@@ -0,0 +1,157 @@
+  width: 100%;
+  border: 1px solid ${({ theme }) => theme.border.color.medium};
+  border-bottom-left-radius: ${({ theme }) => theme.border.radius.sm};
+  border-top-left-radius: ${({ theme }) => theme.border.radius.sm};
+  border-right: none;
+  box-sizing: border-box;
+  background-color: ${({ theme }) => theme.background.transparent.lighter};
+  overflow: hidden;
+  padding: ${({ theme }) => theme.spacing(2)};
+
+  .editor-content {
+    width: 100%;
+  }
+
+  .tiptap {
+    display: flex;
+    align-items: center;
+    height: 100%;
+    color: ${({ theme }) => theme.font.color.primary};
+    font-family: ${({ theme }) => theme.font.family};
+    font-weight: ${({ theme }) => theme.font.weight.regular};
+    border: none !important;
+    white-space: nowrap;
+
+    p.is-editor-empty:first-of-type::before {
+      content: attr(data-placeholder);
+      color: ${({ theme }) => theme.font.color.light};
+      float: left;
+      height: 0;
+      pointer-events: none;
+    }
+
+    p {
+      margin: 0;
+    }
+
+    .variable-tag {
+      color: ${({ theme }) => theme.color.blue};
+      background-color: ${({ theme }) => theme.color.blue10};
+      padding: ${({ theme }) => theme.spacing(1)};
+      border-radius: ${({ theme }) => theme.border.radius.sm};
+    }
+  }
+
+  .ProseMirror-focused {
+    outline: none;
+  }
+`;
+
+interface VariableTagInputProps {
+  inputId: string;
+  label?: string;
+  value?: string;
+  onChange?: (content: string) => void;
+  placeholder?: string;
+}
+
+export const VariableTagInput = ({
+  inputId,
+  label,
+  value,
+  placeholder,
+  onChange,
+}: VariableTagInputProps) => {
+  const editor = useEditor({
+    extensions: [
+      Document,
+      Paragraph,
+      Text,
+      getVariableTag(),
+      Placeholder.configure({
+        placeholder,
+      }),
+    ],
+    editable: true,
+    onCreate: ({ editor }) => {
+      if (isDefined(value)) {
+        initializeEditorContent(editor, value);
+      }
+    },
+    onUpdate: ({ editor }) => {
+      const jsonContent = editor.getJSON();
+      const parsedContent = parseEditorContent(jsonContent);
+      onChange?.(parsedContent);
+    },
+  });
+
+  const insertVariableTag = (variable: string) => {
+    // @ts-expect-error - insertVariableTag is missing from the EditorCommands type but it is defined in the VariableTag extension
+    editor.commands.insertVariableTag(variable);
+  };
+
+  if (!editor) {
+    return null;
+  }
+
+  return (
+    <StyledContainer>
+      {label && <StyledLabel>{label}</StyledLabel>}
+      <StyledInputContainer>
+        <StyledEditor>
+          <EditorContent className="editor-content" editor={editor} />
+        </StyledEditor>
+        <StyledSearchVariablesDropdownContainer>
+          <SearchVariablesDropdown
+            inputId={inputId}
+            onSelect={(variable: string) => insertVariableTag(variable)}
+          />
+        </StyledSearchVariablesDropdownContainer>
+      </StyledInputContainer>
```

Comment:
Tiptap recommends extracting the editor in a separate component to avoid causing too many components to re-render. We might want to move the Dropdown elsewhere.

See https://tiptap.dev/docs/guides/performance#react-tiptap-editor-integration.